airbyte-db/lib/src/main/java/io/airbyte/db/instance/jobs/migrations/V0_35_40_001__MigrateFailureReasonEnumValues.java

```diff
@@ -0,0 +1,557 @@
+    }
+
+    public enum FailureOrigin {
+
+      UNKNOWN("unknown"),
+      SOURCE("source"),
+      DESTINATION("destination"),
+      REPLICATION("replication"),
+      REPLICATION_WORKER("replicationWorker"),
+      PERSISTENCE("persistence"),
+      NORMALIZATION("normalization"),
+      DBT("dbt");
+      private final String value;
+      private final static Map<String, FailureReasonForMigration.FailureOrigin> CONSTANTS = new HashMap<String, FailureReasonForMigration.FailureOrigin>();
+
+      static {
+        for (final FailureReasonForMigration.FailureOrigin c: values()) {
+          CONSTANTS.put(c.value, c);
+        }
+      }
+
+      private FailureOrigin(final String value) {
+        this.value = value;
+      }
+
+      @Override
+      public String toString() {
+        return this.value;
+      }
+
+      @JsonValue
+      public String value() {
+        return this.value;
+      }
+
+      @JsonCreator
+      public static FailureReasonForMigration.FailureOrigin fromValue(final String value) {
+        final FailureReasonForMigration.FailureOrigin constant = CONSTANTS.get(value);
+        if (constant == null) {
+          throw new IllegalArgumentException(value);
+        } else {
+          return constant;
+        }
+      }
+
+    }
+
+
+    public enum FailureType {
+
+      UNKNOWN("unknown"),
+      CONFIG_ERROR_OLD("configError"),
+      SYSTEM_ERROR_OLD("systemError"),
+      MANUAL_CANCELLATION_OLD("manualCancellation"),
+      CONFIG_ERROR("config_error"),
+      SYSTEM_ERROR("system_error"),
+      MANUAL_CANCELLATION("manual_cancellation");
+      private final String value;
+      private final static Map<String, FailureReasonForMigration.FailureType> CONSTANTS = new HashMap<String, FailureReasonForMigration.FailureType>();
+
+      static {
+        for (final FailureReasonForMigration.FailureType c : values()) {
+          CONSTANTS.put(c.value, c);
+        }
+      }
+
+      private FailureType(final String value) {
+        this.value = value;
+      }
+
+      @Override
+      public String toString() {
+        return this.value;
+      }
+
+      @JsonValue
+      public String value() {
+        return this.value;
+      }
+
+      @JsonCreator
+      public static FailureReasonForMigration.FailureType fromValue(final String value) {
+        final FailureReasonForMigration.FailureType constant = CONSTANTS.get(value);
+        if (constant == null) {
+          throw new IllegalArgumentException(value);
+        } else {
+          return constant;
+        }
+      }
+
+    }
+
+  }
+
+
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  @JsonPropertyOrder({
+      "failures",
+      "partialSuccess"
+  })
+  static
+  class AttemptFailureSummaryForMigration implements Serializable {
```

Comment:
These classes will still work even if new enum values or new json properties are added to the db, right? I'm guessing any new values would be ignored?

Or maybe that shouldn't matter because this migration will not be ran after changes are added in future versions?

Reply:
Good call, I should write a test case for what happens if the DB contains a value that isn't known by this class. Even if it _shouldn't_ matter according to changes being added in future versions, it should be possible to make the migration resilient to it.